packages/berry-parsers/sources/syml.ts

```diff
@@ -86,7 +89,11 @@ export function stringifySyml(value: any) {
 
 export function parseSyml(source: string) {
   try {
-    return parse(source.endsWith(`\n`) ? source : `${source}\n`);
+    try {
+      return safeLoad(source) as {[key: string]: any};
```

Comment:
Is there a way to detect a yarn v1 lockfile vs a v2? A nested try catch seems like it would make it difficult to debug if parsing issues are opened.

Reply:
I don't think we can 🙁

The best we could do would be to always parse in strict Yaml, except when we detect a certain comment at the top of the file (like `# yarn v1` and/or `# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY` which was added everywhere back in the v1). That doesn't seem very intuitive however; I'd prefer the migration path to be as simple as possible.

Reply:
Welp, in the end I still have to resort to a check like this because the following is valid Yaml:

```yaml
no-deps@*:
  version "1.0.0"
  resolved "https://registry.yarnpkg.com/no-deps/-/no-deps-1.0.0.tgz#39453512f8241e2d20307975e8d9eb6314f7bf61"
  integrity sha1-OUU1EvgkHi0gMHl16NnrYxT3v2E=
```

Except that it's an object whose value is a string ... 😣